src/animation.rs

```diff
@@ -0,0 +1,503 @@
+//! Animation system.
+//!
+//! ## Introduction
+//!
+//! The `three` animation system is designed around three structures, namely
+//! [`Action`], [`Clip`], and [`Mixer`].
+//!
+//! ### Action
+//!
+//! An [`Action`] controls the playback properties of an animation.
+//! Methods such as [`play`], [`pause`], and [`disable`] are provided to control
+//! an single animation at runtime.
+//!
+//! Actions must be created and updated by a [`Mixer`].
+//!
+//! ### Mixer
+//!
+//! An animation [`Mixer`] schedules the playback of actions.
+//!
+//! The user is expected to create actions from a mixer with the [`Mixer::action`]
+//! function and update actions with the [`Mixer::update`] function.
+//!
+//! ### Clip
+//!
+//! An animation [`Clip`] defines the keyframes and target of an animation.
+//! Clips are usually imported from 3D formats such as glTF.
+//!
+//! ## Walkthrough
+//!
+//! ### Creating a mixer
+//!
+//! First, we create a [`Mixer`] to play our animation.
+//!
+//! ```rust,no_run
+//! // Initialization omitted.
+//! let mut mixer = three::animation::Mixer::new();
+//! # let _ = mixer;
+//! ```
+//!
+//! ### Loading some animation clips
+//!
+//! Now, we load some clips from an animated glTF scene.
+//!
+//! ```rust,no_run
+//! # let mut window = three::Window::builder("", "").build();
+//! let (group, _, _, clips) = window.factory.load_gltf("AnimatedScene.gltf");
+//! window.scene.add(&group);
+//! # let _ = clips;
+//! ```
+//!
+//! ### Creating animation actions
+//!
+//! Now, we schedule the playback of the clips by creating actions.
+//!
+//! The created actions are enabled by default in the 'play' state. This means that
+//! when calling [`Mixer::update`] the created actions will begin to be played back
+//! immediately.
+//!
+//! ```rust,no_run
+//! # let mut window = three::Window::builder("", "").build();
+//! # let mut mixer = three::animation::Mixer::new();
+//! # let (group, _, _, clips) = window.factory.load_gltf("AnimatedScene.gltf");
+//! # window.scene.add(&group);
+//! let actions: Vec<three::animation::Action> = clips
```

Comment:
no need to specify the type here, I prefer doing `.collect::<Vec<_>>()` at the end instead

Reply:
The intention is to be explicit as this is documentation.